### ISSUE TITLE
Cannot change Visibility when applying a Team Template

### ISSUE THREAD

**Issue by m365ken** [microsoftgraph/microsoft-graph-docs#8127]
Hello, When attempting to create a new Team with any Team template (Ex. Standard, HealthCare-hospital, etc ) using the graph api, I'm unable to override the Visibility setting to HiddenMembership. For most templates, I can override visibility to either Public or Private but not HiddenMembership. I get an error stating incorrect value for visibility. The error actually indicates HiddenMembership as a valid value for visibility so it seems like a bug. We would like to begin using Team Templates for new teams but we have a requirement that all Teams be created with HiddenMembership visibility. Right now we achieve this by creating the O365 group first with HiddenMembership visibility then promote that group to a Team. However, we cannot utilize Team templates with this method. Thx

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bf38bbfb-04cd-17ae-b61e-371816ab987f
* Version Independent ID: c2165d6a-a317-6292-32e6-ad67b4fd3faf
* Content: [Create team - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/team-put-teams?view=graph-rest-1.0&tabs=http)
* Content Source: [api-reference/v1.0/api/team-put-teams.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/team-put-teams.md)
* Product: **microsoft-teams**
* Technology: **microsoft-graph**
* GitHub Login: @nkramer
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by JarbasHorst** [comment]
Hello @m365ken ,

as far as I know, providing a value different than private or public for Team's visibility is not supported. 

Regarding your topic: "Right now we achieve this by creating the O365 group first with HiddenMembership visibility then promote that group to a Team. However, we cannot utilize Team templates with this method"
It is possible to define Team templates as well as configuration during teamification. Have a look at [this](https://docs.microsoft.com/en-us/graph/api/team-post?view=graph-rest-beta#example-3-create-a-team-with-multiple-channels-installed-apps-and-pinned-tabs-using-delegated-permissions) documentation for more information.

I hope it helps.

Regards,
Jarbas

**Comment by m365ken** [comment]
Hey @JarbasHorst 
Thanks for pointing me in the right direction. I did try promoting an existing group to a team using a template but got errors. Turns out it was failing as I was using the v1 endbpoint and not beta. I just tested the beta and it works perfectly. The issue with creating a new team with a template and setting the visibility to hiddenmembership still exists. However, your suggestion will be a good work around. Thanks!
Ken

@nkramer 
I totally missed [this ](https://docs.microsoft.com/en-us/graph/api/team-post?view=graph-rest-beta#example-4-create-a-team-from-group). It would have been helpfull if this section was also located [here](https://docs.microsoft.com/en-us/graph/api/team-put-teams?view=graph-rest-beta&tabs=http)
Thx
